### PR TITLE
Add Hippocorpus Dataset

### PR DESCRIPTION
**Pull request by manandey**

### CODE REVIEW

File: datasets/hippocorpus/README.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+multilinguality: 
+- monolingual
+size_categories: 
+- 1K<n<10K
+source_datasets: 
+- original
+task_categories: 
+- text-retrieval
+task_ids: 
+- document-retrieval
```
Comment (lhoestq) [inline datasets/hippocorpus/README.md:19]:
```suggestion
task_categories:
- text-scoring
task_ids:
- text-scoring-other-narrative-flow
```

File: datasets/hippocorpus/README.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+- **Paper:** [Recollection versus Imagination: Exploring Human Memory and Cognition via Neural Language Models](http://erichorvitz.com/cognitive_studies_narrative.pdf)
+- **Point of Contact:** [Eric Horvitz](mailto:horvitz@microsoft.com)
+
+
+### Dataset Summary
+ 
+To examine the cognitive processes of remembering and imagining and their traces in language, we introduce Hippocorpus, a dataset of 6,854 English diary-like short stories about recalled and imagined events. Using a crowdsourcing framework, we first collect recalled stories and summaries from workers, then provide these summaries to other workers who write imagined stories. Finally, months later, we collect a retold version of the recalled stories from a subset of recalled authors. Our dataset comes paired with author demographics (age, gender, race), their openness to experience, as well as some variables regarding the author's relationship to the event (e.g., how personal the event is, how often they tell its story, etc.).
+
+across the languages.
+
```
Comment (lhoestq) [inline datasets/hippocorpus/README.md:60]:
```suggestion
```

**Review by lhoestq — COMMENTED** [review]
Really cool thanks !

Before we merge can you try to reduce the size of the dummy_data.zip file ?

To do so feel free to only keep a few lines of the csv files ans also remove unnecessary chunks of texts (for example keep only the first sentences of a story).

**Comment by manandey** [comment]
> Before we merge can you try to reduce the size of the dummy_data.zip file ?
> 
> To do so feel free to only keep a few lines of the csv files ans also remove unnecessary chunks of texts (for example keep only the first sentences of a story).

Hi @lhoestq, I have reduced the size of the dummy_data.zip file by making the necessary changes you had suggested.

File: datasets/hippocorpus/README.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+multilinguality: 
+- monolingual
+size_categories: 
+- 1K<n<10K
+source_datasets: 
+- original
+task_categories:
+- text-retrieval
+task_ids:
+- document-retrieval
```
Comment (lhoestq) [inline datasets/hippocorpus/README.md:19]:
```suggestion
task_categories:
- text-scoring
task_ids:
- text-scoring-other-narrative-flow
```

File: datasets/hippocorpus/hippocorpus.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                {
+                    # These are the features of your dataset like images, labels ...
+                    "AssignmentId": datasets.Value("string"),
+                    "WorkTimeInSeconds": datasets.Value("string"),
+                    "WorkerId": datasets.Value("string"),
+                    "annotatorAge": datasets.Value("string"),
+                    "annotatorGender": datasets.Value("string"),
+                    "annotatorRace": datasets.Value("string"),
+                    "distracted": datasets.Value("string"),
+                    "draining": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/hippocorpus/hippocorpus.py:77]: Could this be integers instead of strings ?

**Review by lhoestq — COMMENTED** [review]
Nice thanks ! The dummy data are all good now :)

I left a comment about the feature type for certain columns.
In particular I feel like some columns could be integers instead of strings (the ones that are `5-point Likert`)

File: datasets/hippocorpus/hippocorpus.py (hunk 4 above)
Comment (manandey) [reply datasets/hippocorpus/hippocorpus.py]: Hi @lhoestq, I have changed all the (5-point Likert) columns from string to float. A few checks have failed due to` FAILED tests/test_file_utils.py::TempSeedTest::test_tensorflow` .

**Review by lhoestq — APPROVED** [review]
Looks all good now ! thanks :)

**Comment by lhoestq** [comment]
merging since the CI is fixed on master

**Pull request merged**